Documentation/kubernetes-upgrade.md

```diff
@@ -0,0 +1,68 @@
+# Upgrading Kubernetes
+
+This document will describe upgrading your cluster's master and worker nodes. For general information on Kubernetes cluster management, upgrades (including more advanced topics such as major API version upgrades) see the [Kubernetes upstream documentation](http://kubernetes.io/v1.0/docs/admin/cluster-management.html).
+
+## Upgrading the Kubelet
+
+The Kubelet binary ships with CoreOS releases. As the host OS is updated, the Kubelet will be upgraded as well.
+
+If you wish to run a custom version of the kubelet binary, the kubelet service file (`/etc/systemd/system/kubelet.service`) should be modified to use the custom binary path on each node (master & worker).
+
+## Upgrading Master Nodes
+
+Master nodes will consist of the following Kubernetes components:
+
+* kubelet
+* kube-proxy
+* kube-apiserver
+* kube-controller-manager
+* kube-scheduler
+* kube-podmaster (high-availability)
+
+### Upgrading kube-apiserver and kube-proxy
+
+Both the kube-apiserver and kube-proxy are run as "static pods". This means the pod definition is a file on disk (default location: `/etc/kubernetes/manifests`). To update these components, you simply need to update the static manifest file.
+
+For example, to upgrade the kube-apiserver version, you could make the following change to: `/etc/kubernetes/manifests/kube-apiserver.yaml`
+
+```
+# In the pod manifest, change the hyperkube image version from:
+    image: gcr.io/google_containers/hyperkube:v1.0.6
+# To:
+    image: gcr.io/google_containers/hyperkube:v1.0.7
+```
+
+*NOTE:* If you are running a multi-master high-availabililty cluster, please see the next section on upgrading the remaining master components. Otherwise you can upgrade the remaining static pods (controller-manager, scheduler) using the same process described above.
+
+### Upgrading Remaining Master Components (high-availability)
+
+The kube-controller-manager, kube-scheduler, and kube-podmaster are all also deployed as static pods in `/etc/kubernetes/manifests`. However, in high-availability deployments, the kube-podmaster is responsible for making sure only a single copy of the controller-manager and scheduler are running cluster-wide.
+
+To accomplish this the kube-podmaster on each node, if master-elected, will copy the static manifest from `/srv/kubernetes/manifets` into `/etc/kubernetes/manifests` and the kubelet will pick up the manifest and run the pod. If the kube-podmaster loses its status as master, it will remove the static pod from `/etc/kubernetes/manifests/` and the kubelet will shut down the pod.
+
+This configuration means upgrading of these components will take a little more coordination.
+
+To upgrade the kube-controller-manager and kube-scheduler:
+
+1. Make changes to the base manifests in `/srv/kubernetes/manifests`
```

Comment:
The manifests should be changed in all masters, active or not (so if master changes, it's still  using updated manifest). But I can make a note about this.